React-Dojo/react_typescript/src/App.tsx

```diff
@@ -0,0 +1,29 @@
+import './App.css';
+import {
+  Routes,
+  Route,
+  Navigate
+} from "react-router-dom";
+
+import ClassBased from './pages/ClassBased';
+import Functional from './pages/Functional';
+import MainNavigation from './components/MainNavigation';
+
+const App: React.FC = () => {
+
+  const API_URL = 'https://gateway.marvel.com/v1/public/characters'
+  const API_KEY = '&apikey=7fb70a9709792d94b28b122a317fe87f'
```

Comment:
Your API key is exposed on github, so others can access and use it. Try to save it in a .env file and use it from there. Also don't forget to add the .env file to gitignore.